gTea/app/src/main/java/com/nechaieva/gtea/fragments/MenuFragment.java

```diff
@@ -112,12 +115,15 @@ void checkVoiceOrder() {
             Log.e(ORDER_TAG, "No MainActivity found");
             return;
         }
+
         Bundle order = mainActivity.getStartingInfo();
+        Optional<String> menuItem = checkInMenu(order);
```

Comment:
It feels like this code could be refactored to simplify it.

Is the variable "navigateTo" required?

Proposal:

if (!menuItem.IsPresent()) {
   // some logging
  return;
}
order.putString(DeepLink.ORDER_BUNDLE_TAG, menuItem.get());
navigateToOrder(order); // put the success logging into this method